### PR TITLE
Closes #185 - SpringBoot-Actuator based C8-Health

### LINKED ISSUE #185: C8: Extend Health-Checks with Job-Worker-Health
## Description

The Actuator-based Health-Checks for the C7 implementation provide a lot of insight into the health of the Adapter.
Extend it for C8.
Similar to C7, assume the support of multiple C8 instances and composite health indicators accordingly.

## Definition of Done

- [x] Added Health-Checks for all JobWorkers
- [x] Tested C8 Health-Checks similar to C7

### PR DESCRIPTION
**Pull request by bruderj15**
## Definition of Done

- [x] The corresponding ticket is in state `In Review`
- [x] The corresponding ticket is attached to this Pull-Request
- [x] The commit-message-format `Closes #ISSUE_ID - PROBLEM/SOLUTION`
    - is present as single-commit already or
    - will be squashed on merge
- [ ] The changes pass the SonarQubeCloud-Quality-Gate
- [x] If the documentation needs an update, following there's a link to it: https://github.com/kadai-io/kadai-doc/pull/140
- [x] If extra release-notes are required, they're listed indented below:

### CODE REVIEW

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/config/health/Camunda8HealthConfigurationProperties.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public CompositeHealthContributorConfigurationProperties getComplete() {
+      return complete;
+    }
+
+    public void setComplete(CompositeHealthContributorConfigurationProperties complete) {
+      this.complete = complete;
+    }
+
+    public JobWorkerHealthConfigurationProperties withComplete(
```
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/config/health/Camunda8HealthConfigurationProperties.java:39]: Are the withX functions necessary, since we have already the setter?

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.HealthContributor;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Camunda8HealthContributorFactory implements PluginHealthContributorFactory {
+
+  private final Camunda8HealthConfigurationProperties properties;
+  private final Camunda8System camunda8System;
```
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java]: This is never used, right?

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicator.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+public class Camunda8JobWorkerHealthIndicator implements HealthIndicator {
+  private final MonitoredComponent monitoredComponent;
+
+  public Camunda8JobWorkerHealthIndicator(MonitoredComponent monitoredComponent) {
+    this.monitoredComponent = monitoredComponent;
+  }
+
+  @Override
+  public Health health() {
+    Instant lastRun = monitoredComponent.getLastRun().getEnd();
```
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicator.java:20]: What happens here if job hasnt started but we want to check the health, won't a NullPointerException be thrown here?

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  private final UserTaskCompletion userTaskCompletion;
+  private final UserTaskCreation userTaskCreation;
+
+  @Autowired
+  public Camunda8HealthContributorFactory(
+      Camunda8HealthConfigurationProperties properties,
+      Camunda8System camunda8System,
+      UserTaskCompletion userTaskCompletion,
+      UserTaskCreation userTaskCreation) {
```
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java:23]: Shouldnt we add null check to UserTaskCompletion and UserTaskCreation?

File: kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactoryTest.java
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactoryTest.java:1]: Shouldn't we add cases where the job workers are disabled?

File: kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicatorTest.java
Comment (jamesrdi) [inline kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicatorTest.java:1]: Maybe add a test for the case when getLastRun or getExpectedRunDuration is null.

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/config/health/Camunda8HealthConfigurationProperties.java (hunk 1 above)
Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/config/health/Camunda8HealthConfigurationProperties.java:39]: I think they're fine. They do lead to much more readable tests.

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java (hunk 4 above)
Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java:23]: Why should the beans be null?

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicator.java (hunk 3 above)
Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicator.java:20]: `MonitoredComponent::getLastRun` always returns non-null by convention of the spec.

File: kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicatorTest.java
Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8JobWorkerHealthIndicatorTest.java:1]: Impossible by design

File: kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactoryTest.java
Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/test/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactoryTest.java:1]: This is not in-scope for this class here. It doesn't even know the job-workers.
Also already covered with `Camunda8JobWorkerHealthCompositeTest`.

File: kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java (hunk 4 above)
Comment (jamesrdi) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java:23]: If the Camunda8JobWorkerHealthIndicator receives a null monitoredComponent, wouldn't it throw NPE?

Comment (bruderj15) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java:23]: Again, why would a bean be null?

Comment (jamesrdi) [reply kadai-adapter-camunda-8-system-connector/src/main/java/io/kadai/adapter/systemconnector/camunda/monitoring/Camunda8HealthContributorFactory.java:23]: Ahh I see your point

**Comment by jamesrdi** [comment]
Cant we do something with the Coverage from the SonarCloud analysis?

**Comment by bruderj15** [comment]
> Cant we do something with the Coverage from the SonarCloud analysis?

We can - bot not here. Will improve drastically with #187 and #240.

**Review by jamesrdi — APPROVED** [review]
Looks good to me!
